engine/src/main/java/org/terasology/rendering/dag/RenderTaskListGenerator.java

```diff
@@ -102,41 +115,41 @@ private void logIntermediateRendererListForDebugging(List<Node> orderedNodes) {
 
         for (Node node : orderedNodes) {
             if (node.isEnabled()) {
-
                 if (logger.isInfoEnabled()) {
                     // Marker tasks just add a dividing line to the logger output
                     taskList.add(new MarkerTask(node.getClass().getSimpleName()));
                     enabledNodes++; // we count them only for statistical purposes
                     potentialTasks += 2 * node.getDesiredStateChanges().size() + 1;
                 }
 
-                // generating tasks for the desired state changes
+                // Examine the StateChanges requested by current Node, and insert/update them in persistentStateChanges.
                 for (StateChange currentStateChange : node.getDesiredStateChanges()) {
+                    StateChange persistentStateChange = persistentStateChanges.get(currentStateChange.getClass());
```

Comment:
Regarding the first point: the persistentStateChange is used ONCE in the entire method, and that is here. It can technically be removed by making the next line `if (!currentStateChange.equals(persistentStateChange))` much longer, but I prefered not doing that. With that being said, I really don't think the type of a variable is the kind of redundant information that needs to be removed, at the expense of increasing the scope of a variable unneccesarily. When I originally encountered this code, I was confused why this variable was defined at top and not locally.
Can you please consider letting the type information remain here? Pretty please? :D